### PR TITLE
[PR] partial neutral recycling source

### PR DESCRIPTION
**Pull request by Antoinehoff**
This PR addresses DR #1010 _and did not require the use of any coding agent._

We have introduced two new parameters to the adaptive source model, allowing for partial compensation of energy and particle losses. This provides finer control over the source's response compared to the previous binary (all-or-nothing) adaptation.
We also write explicitly the recycling factor in the TCV adapt source regression tests to display the feature.

### Implementation Details
Two double fields were added to the gkyl_gyrokinetic_adapt_source structure to represent the recycling fractions:

```C
struct gkyl_gyrokinetic_adapt_source {
  // ...
  double adapt_particle_fraction; // Fraction of particle loss to compensate (default: 1.0)
  double adapt_energy_fraction;   // Fraction of energy loss to compensate (default: 1.0)
  // ...
};
```

In `gk_species_sources.c`, these parameters are passed to the `gk_adapt_source` structure. To maintain backward compatibility, the fractions default to 1.0 if they are not explicitly defined in the input table (handling cases where the value is initialized to 0 which is equal to left undefined).

In `gk_species_source_adapt_enabled`, the calculated particle and energy loss rates are now scaled by these fractions before being added to the source input:

```C
double density_compensation = adapt_src->adapt_particle_fraction * adapt_src->particle_rate_loss;
double particle_src_new = adapt_src->adapt_particle ? 
    particle_input + density_compensation : particle_input;
```

### Verification
We verify the implementations using `rt_gk_tcv_iwl_adapt_source_2x2v_p1.c`. We run 2 step and print the source adaptation details.

#### 100% recycling (fraction is left undefined)
At initialization phase:
```
  Setting up adaptive source 1 for species elc: 
  adapt_particle = 1, adapt_energy = 0, 
  adapt_particle_fraction = 1, adapt_energy_fraction = 1
```
After step 2:
```
  Adaptive source 1 for species elc: 
  particle_input = 0, energy_input = 0, 
  particle_loss = 5.64376e+21, energy_loss = 187659, 
  particle_src_new = 5.64376e+21, energy_src_new = 0
```
We see that the source particle injection matches the particle losses.

#### 80% recycling (set fraction to be 0.8)
At initialization phase:
```
  Setting up adaptive source 1 for species elc: 
  adapt_particle = 1, adapt_energy = 0, 
  adapt_particle_fraction = 0.8, adapt_energy_fraction = 1
```
After step 2:
```
  Adaptive source 1 for species elc: 
  particle_input = 0, energy_input = 0, 
  particle_loss = 5.64376e+21, energy_loss = 187659, 
  particle_src_new = 4.51501e+21, energy_src_new = 0
```
The 80% fraction worked and we see that the new particle injection rate of the source is 4.51501/ 5.64376=0.8000003544 times smaller than the losses.

### CODE REVIEW

File: gyrokinetic/apps/gk_species_source.c [hunk 1]
```diff
@@ -455,6 +456,10 @@ gk_species_source_init(struct gkyl_gyrokinetic_app *app, struct gk_species *s,
 
         adapt_src->adapt_particle = s->info.source.adapt[k].adapt_particle;
         adapt_src->adapt_energy = s->info.source.adapt[k].adapt_energy;
+        adapt_src->adapt_particle_fraction = s->info.source.adapt[k].adapt_particle_fraction > 0?
```
Comment (manauref) [inline gyrokinetic/apps/gk_species_source.c]: in general, when the input is an optional double, the best practice is to also have a boolean input like
`has_adapt_particle_fraction = true,`
see what we did for x_LCFS in the geometry table. This is because adapt_particle_fraction=0 is a technically valid input, or one may wish to run with `adapt_particle_fraction=0` for testing/dev

**Review by manauref — APPROVED** [review]
approving, just left light suggestion below

File: gyrokinetic/apps/gk_species_source.c (hunk 1 above)
Comment (Antoinehoff) [reply gyrokinetic/apps/gk_species_source.c]: Ok I see the idea, I will add this.

**Review by tnbernard — APPROVED** [review]
